Mathlib/Topology/Category/CompHaus/ExplicitLimits.lean

```diff
@@ -172,6 +219,94 @@ def finiteCoproduct.isColimit : Limits.IsColimit (finiteCoproduct.cocone X) wher
     apply_fun (fun q => q t) at hm
     exact hm
 
+section Iso
+
+noncomputable
+def ToFiniteCoproduct : ∐ X ⟶ finiteCoproduct X :=
+  Limits.Sigma.desc <| fun a => finiteCoproduct.ι X a ≫ (𝟙 _)
+
+noncomputable
+def FromFiniteCoproduct : finiteCoproduct X ⟶ ∐ X :=
+  finiteCoproduct.desc X <| fun a => (Limits.Sigma.ι X a)
+
+@[simp]
+theorem toFiniteCoproduct_comp_fromFiniteCoproduct_eq_id :
+    (ToFiniteCoproduct X ≫ FromFiniteCoproduct X) = 𝟙 _ := by
+  ext
+  simp [ToFiniteCoproduct, FromFiniteCoproduct]
+
+@[simp]
+theorem fromFiniteCoproduct_comp_toFiniteCoproduct_eq_id :
+    (FromFiniteCoproduct X ≫ ToFiniteCoproduct X) = 𝟙 _ := by
+  refine' finiteCoproduct.hom_ext _ _ _ (fun a => _)
+  simp [ToFiniteCoproduct, FromFiniteCoproduct]
+
+@[simps]
+noncomputable
+def FromFiniteCoproductIso : finiteCoproduct X ≅ ∐ X where
+  hom := FromFiniteCoproduct X
+  inv := ToFiniteCoproduct X
+  hom_inv_id := fromFiniteCoproduct_comp_toFiniteCoproduct_eq_id X
+  inv_hom_id := toFiniteCoproduct_comp_fromFiniteCoproduct_eq_id X
+
+@[simps]
+noncomputable
+def ToFiniteCoproductIso : ∐ X ≅ finiteCoproduct X where
+  hom := ToFiniteCoproduct X
+  inv := FromFiniteCoproduct X
+  hom_inv_id := toFiniteCoproduct_comp_fromFiniteCoproduct_eq_id X
+  inv_hom_id := fromFiniteCoproduct_comp_toFiniteCoproduct_eq_id X
+
+theorem isIso_toFiniteCoproduct :
+    IsIso (ToFiniteCoproduct X) :=
+  ⟨FromFiniteCoproduct X, by simp, by simp⟩
+
+theorem isIso_fromFiniteCoproduct :
+    IsIso (FromFiniteCoproduct X) :=
+  ⟨ToFiniteCoproduct X, by simp, by simp⟩
+
+@[simp]
+theorem Sigma.ι_comp_toFiniteCoproduct (a : α) :
+    (Limits.Sigma.ι X a) ≫ ToFiniteCoproduct X = finiteCoproduct.ι X a := by
+  simp [ToFiniteCoproduct]
+
+@[simp]
+theorem finiteCoproduct.ι_comp_fromFiniteCoproduct (a : α) :
+    (finiteCoproduct.ι X a) ≫ FromFiniteCoproduct X = Limits.Sigma.ι X a := by
+  simp [FromFiniteCoproduct]
+
+@[simps] noncomputable
+def ToFiniteCoproductHomeo : (∐ X : _) ≃ₜ finiteCoproduct X where
```

Comment:
Don't we have some general way of converting isomorphisms in `CompHaus` to homeomorphisms? Maybe one has to pass through `TopCat`?